### ISSUE TITLE
TestLibrary and createLibrary not found

### ISSUE THREAD

**Issue by binkley** [unbroken-dome/gradle-testsets-plugin#64]
I'm trying the documented `TestLibrary` feature, and Gradle is not finding that class, nor is the `createLibrary` method found; the suggested `import` statement is rejected as well.  Output from Gradle is:
```
> class org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestSet_Decorated cannot be cast to class org.unbrokendome.gradle.plugins.testsets.dsl.TestLibrary (org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestSet_Decorated and org.unbrokendome.gradle.plugins.testsets.dsl.TestLibrary are in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @5c8fd451)
```
IntelliJ highlights "TestLibrary" in `import org.unbrokendome.gradle.plugins.testsets.dsl.TestLibrary` in red (not found).

I'm using Java 11, Gradle 5.1, and test-sets 2.0.3.

**Comment by binkley** [comment]
I worked out that I needed to add the plugin as a dependency into `buildscript`.  The IDE still has difficulty, but command line Gradle is better.  Perhaps a documentation update for the `TestLibrary` example?

However, using:
```
testSets {
    testCommon(TestLibrary)

    unitTest {
        imports testCommon
    }

    integrationTest {
        imports testCommon
        shouldRunAfter test
    }
}
```
I now see this from command line:
```
class org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestSet_Decorated cannot be cast to class org.unbrokendome.gradle.plugins.testsets.dsl.TestLibrary (org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestSet_Decorated and org.unbrokendome.gradle.plugins.testsets.dsl.TestLibrary are in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @81e7b6c)
```
Removing `shouldRunAfter test` fixes this complaint while keeping the `imports`, and removing the `imports` lets Gradle process `shouldRunAfter`.

**Comment by tkrullmann** [comment]
Ok, this is some really strange behavior :-)

First, the Groovy DSL support in IntelliJ isn't very good in recognizing types from plugins - you can use the `createLibrary` method to avoid the import statement:

```
testSets {
    def testCommon = createLibrary('testCommon')
}
```

Second, there is no `shouldRunAfter` method on a test set (although it might be a good feature to add). Apparently by using this method, Gradle figures out that you mean to configure the `integrationTest` task instead of the `integrationTest` testset - remember that tasks are available as top-level variables when using Groovy DSL.

This in turn causes the `imports` method to be called on the container, where it exists as a Kotlin extension method so that Kotlin scripts can refer to libraries by name. However, Groovy doesn't understand extension methods, so this gets messed up (and there's also a bug in the method).

To fix the problem, please take the `shouldRunAfter` call outside of `testSets` so it properly refers to the `integrationTest` task:

```groovy
testSets {
    // ...

    integrationTest {
        imports testCommon
    }
}

integrationTest.shouldRunAfter test
```

**Comment by tkrullmann** [comment]
Should be fixed with 2.1.0.

The error message for your example is now clearer:
  `Could not find method imports() for arguments [org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestLibrary_Decorated@71edb2dd] on task ':integrationTest' of type org.gradle.api.tasks.testing.Test.`

(At least now it's obvious that you're trying to configure the `integrationTest` task and not the test set.)

I also introduced a new `libraries` block which should make it easier to create libraries:

```groovy
testSets {
    libraries {
        testCommon
    }

    unitTest {
        imports testCommon
    }

    integrationTest {
        imports testCommon
    }
}
```
